modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1917,6 +1927,14 @@ void GDScriptAnalyzer::resolve_assignable(GDScriptParser::AssignableNode *p_assi
 		}
 	}
 
+#ifdef DEBUG_ENABLED
+	if (GLOBAL_GET("debug/gdscript/warnings/enforce_static_typing") && type.type_source != GDScriptParser::DataType::ANNOTATED_INFERRED) {
+		if (!p_assignable->datatype_specifier && !(type.kind == GDScriptParser::DataType::BUILTIN && type.builtin_type == Variant::NIL)) {
```

Comment:
Why is there an exception for type `null`/`void` (`Nil`)? You can't specify it explicitly and most likely implicitly either.